platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java

```diff
@@ -0,0 +1,224 @@
+    public static final String PROPERTY_LOGGING_HANDLER_LEVEL = "logging.handler.%s.level";
+    public static final String PROPERTY_PACKAGE_LEVEL = "%s.";
+    public static final String PROPERTY_LOGGING_MARKER = "logging.marker";
+    public static final String PROPERTY_LOGGING_HANDLER_MARKER = "logging.handler.%s.marker";
+
+    /**
+     * The emergency logger.
+     */
+    private static final EmergencyLogger EMERGENCY_LOGGER = EmergencyLoggerProvider.getEmergencyLogger();
+
+    /**
+     * The cache for the levels.
+     */
+    private final Map<String, ConfigLevel> levelCache;
+
+    /**
+     * The cache for the markers.
+     */
+    private final Map<String, MarkerDecision> markerCache;
+
+    /**
+     * The configuration properties.
+     */
+    private final Map<String, ConfigLevel> levelConfigProperties;
+
+    /**
+     * The prefix for the configuration.
+     */
+    private final String name;
+
+    /**
+     * Creates a new logging level configuration based on the given configuration and handler name.
+     *
+     * @param configuration The configuration.
+     * @param name        The name.
+     */
+    public HandlerLoggingLevelConfig(@NonNull Configuration configuration, @NonNull String name) {
+        this.name = Objects.requireNonNull(name, "name must not be null");
+        this.levelCache = new ConcurrentHashMap<>();
+        this.markerCache = new ConcurrentHashMap<>();
+        this.levelConfigProperties = new ConcurrentHashMap<>();
+        update(configuration);
+    }
+
+    /**
+     * Updates the configuration based on the given configuration. That method can be used to change the logging level
+     * dynamically at runtime.
+     *
+     * @param configuration The configuration.
+     */
+    public void update(final @NonNull Configuration configuration) {
+        Objects.requireNonNull(configuration, "configuration must not be null");
+
+        final String propertyHandler = PROPERTY_LOGGING_HANDLER_LEVEL.formatted(name);
+        final ConfigLevel defaultLevel =
+                configuration.getValue(PROPERTY_LOGGING_LEVEL, ConfigLevel.class, ConfigLevel.UNDEFINED);
+        final ConfigLevel defaultHandlerLevel =
+                configuration.getValue(propertyHandler, ConfigLevel.class, ConfigLevel.UNDEFINED);
+
+        levelConfigProperties.clear();
+        if (defaultLevel == ConfigLevel.UNDEFINED && defaultHandlerLevel == ConfigLevel.UNDEFINED) {
+            levelConfigProperties.put("", ConfigLevel.INFO);
+        } else if (defaultHandlerLevel != ConfigLevel.UNDEFINED) {
+            levelConfigProperties.put("", defaultHandlerLevel);
+        } else {
+            levelConfigProperties.put("", defaultLevel);
+        }
+
+        levelConfigProperties.putAll(readLevels(PROPERTY_LOGGING_LEVEL, configuration));
+        levelConfigProperties.putAll(readLevels(propertyHandler, configuration));
+        levelCache.clear();
```

Comment:
Next step: instead of calling clear we can regenerated the values for all available keys?